include/highfive/bits/H5Utils.hpp

```diff
@@ -38,63 +38,90 @@ class FixedLenStringArray;
 
 
 namespace details {
-
-// determine at compile time number of dimensions of in memory datasets
 template <typename T>
-struct array_dims {
-    static constexpr size_t value = 0;
+struct manipulator {
+    using type = T;
+
+    static const size_t n_dims = 0;
+    static const size_t r_n_dims = n_dims;
```

Comment:
To me, the variable names are not really descriptive...

Reply:
Help me with that, please, I'm not good.

Reply:
Some suggestions:

* `n_dims` -> `rank` / `dimension` or `outer_rank` / `outer_dimension` 
* `r_n_dims` -> `recursive_rank` / `total_rank` / `full_rank` / `*_dimension` or `rank` / `dimension`

Or... something even better ;) I'd be happy to iterate.

Reply:
Maybe also
 `manipulator` -> `array_info` ?
Or you intent that in the future to manipulate something? (but the word is not much used I guess)

Reply:
for the properties, I would keep consistency. In DataSpace we have the following methods:
```
size_t getNumberDimensions() const;
size_t getElementCount() const;
```
so, what about
`.number_dimensions`
`.element_count`

Reply:
`n_dims` => number of dimensions of the Object itself (e.g `std::vector` => 1)
`r_n_dims` => total number of dimensions (e.g `std::vector<std::vector>` => 2),
It's not an  `element_count`

Reply:
it's not only `array_info`, i  will add methods to resize array, `ŧransform_read`, `transform_write`, etc...